### PR TITLE
OO fixes after JLab course

### PR DESCRIPTION
**Pull request by bernhardmgruber**
Addresses part of #428

### CODE REVIEW

File: talk/objectorientation/objectsclasses.tex [hunk 1]
```diff
@@ -65,37 +65,58 @@
...
+          MyClass mc;
+          ...
+        }
+      \end{cppcode*}
+    \end{block}
+    \begin{block}{User 2: fun.cpp}
+      \begin{cppcode*}{gobble=4}
+        #include "MyClass.hpp"
+        int fun(MyClass& mc) {
+          return mc.squareA();
```
Comment (hageboeck) [inline talk/objectorientation/objectsclasses.tex:102]: The function returns `void`, so maybe it's better to make it return something.

**Review by hageboeck — CHANGES_REQUESTED** [review]
A few things to fix before we can merge.

File: talk/objectorientation/constructors.tex [hunk 2]
```diff
@@ -224,14 +223,15 @@
...
       };
-      struct DerivedClass : BaseClass {
-        using BaseClass::BaseClass;
-        DerivedClass(int a, int b);
+      struct Derived : Base {
+        using Base::Base;
+        Derived(int a, int b); // 2
       };
-      DerivedClass a{5};
+      Derived d{5};    // calls 1
```
[the remark this answers is not in the document]
Comment (bernhardmgruber) [reply talk/objectorientation/constructors.tex]: Renamed to `ctor 1` etc.

File: talk/objectorientation/objectsclasses.tex (hunk 1 above)
Comment (bernhardmgruber) [reply talk/objectorientation/objectsclasses.tex]: Fixed, made everything return `void`.

File: talk/objectorientation/objectsclasses.tex [hunk 3]
```diff
@@ -116,16 +137,16 @@
...
   \begin{cppcode}
-    void externalFunc(MyStruct & s);
+    void fun(S & s);
 
-    struct MyStruct {
-      void invokeExternalFunc() {
-        externalFunc(*this); // Pass a reference to ourself
+    struct S {
+      void haveFun() {
+        fun(*this); // Pass a reference to ourself
```
[the remark this answers is not in the document]
Comment (bernhardmgruber) [reply talk/objectorientation/objectsclasses.tex]: No, they aren't. I picked different ones.

**Review by hageboeck — APPROVED** [review]
LGTM! 👍

**Comment by bernhardmgruber** [comment]
Thx!

I fixed the typo that codespell reported as well now.

**Pull request merged**